iceoryx_posh/include/iceoryx_posh/popo/client_options.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_CLIENT_OPTIONS_HPP
+#define IOX_POSH_POPO_CLIENT_OPTIONS_HPP
+
+#include "iceoryx_posh/internal/popo/ports/client_server_port_types.hpp"
+#include "iceoryx_posh/popo/port_queue_policies.hpp"
+
+#include "iceoryx_hoofs/cxx/serialization.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace popo
+{
+/// @brief This struct is used to configure the client
+struct ClientOptions
+{
+    /// @brief The size of the response queue where chunks are stored before they are passed to the user
+    /// @attention Depending on the underlying queue there can be a different overflow behavior
+    uint64_t responseQueueCapacity{ClientChunkQueueData_t::MAX_CAPACITY};
+
+    /// @brief The name of the node where the client should belong to
+    iox::NodeName_t nodeName{""};
+
+    /// @brief The option whether the client shall try to connect when creating it
+    bool connectOnCreate{true};
+
+    /// @brief The option whether the server should block when the response queue is full
+    /// @note Corresponds with ServerOptions::clientTooSlowPolicy
+    QueueFullPolicy2 responseQueueFullPolicy{QueueFullPolicy2::DISCARD_OLDEST_DATA};
+
+    /// @brief The option whether the client should block when the request queue is full
+    ConsumerTooSlowPolicy serverTooSlowPolicy{ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA};
+
+    /// @brief serialization of the ClientOptions
+    cxx::Serialization serialize() const noexcept;
```

Comment:
Here I would prefer the casting operator and for the deserialization the conversion ctor. This would have the nice advantage that you could use it directly in a method like `send(cxx::Serialization)` as argument.

Reply:
Sure, but currently it's not used that way in our code base but more like `static_cast<cxx::Serialization>(foo)`. Which is much more to type than `foo.serialize()` ;)

Additionally, there is the symmetry with `deserialize`.

No too strong feelings about this, though, and I can change it if you want. We need to overhaul the serialization at some point in the future, though

Reply:
Yeah lets do it later. Then it is a problem of future elBoberido and future elfenpiff. I really love those guys, they are willing to do so much for us in contrast to past elfenpiff who is so lazy and gives me all of his work ;)

Reply:
yeah, those are two lazy nogoodniks :laughing: